x-pack/plugins/alerting/server/rules_client/methods/create.ts

```diff
@@ -103,7 +103,7 @@ export async function create<Params extends RuleTypeParams = never>(
 
   // TODO https://github.com/elastic/kibana/issues/148414
   // If any action-level frequencies get pushed into a SIEM rule, strip their frequencies
-  const firstFrequency = data.actions[0]?.frequency;
+  const firstFrequency = data.actions.find((action) => action?.frequency)?.frequency;
```

Comment:
Tell if I am wrong here, but @e40pud should be able to delete that completely. I am wondering if it will be better to avoid the change for now.???

Reply:
Yes, after Zhenya's changes this should be removed.

I've made change here to avoid broken tests on Security Solution side.
Since the change of how we are merging existing actions and legacy ones:

> - legacy actions from sidecar SO, now will be merged with rules actions, if there any.
Before, in the previous implementation, there was inconsistency in a way how legacy and rules actions were treated.
  - On read: actions existing in rule, [would take precedence over legacy ones  ](https://github.com/elastic/kibana/blob/8.7/x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/normalization/rule_actions.ts#L94-L114)
  - On migration: SO actions [only saved](https://github.com/elastic/kibana/blob/8.7/x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/rule_actions/legacy_action_migration.ts#L114). If any actions present in rule, they will be lost.


Existing actions do not have frequencies, but migrated already have(introduced in this PR). So, I just made a small amend how frequency is searched in a list of actions. Since, after merging existing and legacy, existing action are first in order, but don't have frequencies to be removed